### DISCUSSION TITLE
Split trailingSlash option into two. One for routing and one for export (subFolder)

### DISCUSSION THREAD

**Discussion by jarandmi** [vercel/next.js#48141] [Ideas]
### Describe the feature you'd like to request

Today, the `trailingSlash` config option behaves so when navigating the app it add a "/" to every route, even if it didn't exist in the link. Thats expected behaviour. What it also does is, when exporting static .html files, it creates a subfolder with index.html files, so the routing is targeting the folder, and not the file.

This are two different needs. When hosting on S3, we need the files in folders, so we can get nice and clean URL without .html. But we don't want the trailing slash. 

### Describe the solution you'd like

`{ exportToSubFolders: true, trailingSlash: false }`

This makes it possible to host files in folder, but still route/navigate to paths without the trailing slash.

### Describe alternatives you've considered

You can try to add rewrite rules to your hosting platform, but adding two different options seems like a way more simpler and logic solution

**Comment by RonnyO** [comment]
I also need this exact split 👍
@jarandmi did you find any workaround?

**Reply by jarandmi** [reply]
@RonnyO I added a 301 redirect in our lambda, if the URL had a trailing slash, so didn't solve it within Next.js

**Reply by jarandmi** [reply]
I also patched Next so it don't build with .html even if trailingSlash is true

**Comment by matthewjamesadam** [comment]
I'm statically exporting my next.js site, and need to export pages with the structure `/path/to/file/index.html`, instead of `/path/to/file.html`.

I've enabled `trailingSlash`, which does export files in this structure, but now all my links have a trailing slash, which I don't want.  I think it would be great to separate these two features out.

**Reply by gavenkoa** [reply]
It is not a feature but a market strategy to force you buy their hosting solution instead of using zero cost generic Web Server.

**Comment by jarandmi** [comment]
I fix this with patching Next package in yarn

```
diff --git a/dist/export/index.js b/dist/export/index.js
index 58615a7ce8ffdcc5a027f8c020e22e35a44bddd3..fe530e1cb8536694234f122fb407aa1d2728dadb 100644
--- a/dist/export/index.js
+++ b/dist/export/index.js
@@ -238,7 +238,7 @@ async function exportAppImpl(dir, options, span) {
             appDir: null
         }));
     }
-    const subFolders = nextConfig.trailingSlash && !options.buildExport;
+    const subFolders = true && !options.buildExport;
     if (!options.silent && !options.buildExport) {
         _log.info(`using build directory: ${distDir}`);
     }
```
